bundles/org.eclipse.ui.workbench/eclipseui/org/eclipse/ui/internal/progress/ProgressManager.java

```diff
@@ -970,6 +970,8 @@ public void run(boolean fork, boolean cancelable, IRunnableWithProgress runnable
 		if (!fork || !cancelable) {
 			// Backward compatible code.
 			final ProgressMonitorJobsDialog dialog = new ProgressMonitorJobsDialog(null);
+			dialog.setOpenOnRun(false);
```

Comment:
I wonder if:
1) The original code should simpy be wrapped into `if(!shouldRunInBackground()) ...`
2) The original code should use `ProgressManagerUtil.getDefaultParent()` instead of `null` on the line above